### ISSUE TITLE
Missing feature - notification window of resources sent to player

### ISSUE THREAD

**Issue by IvanSavenko** [vcmi/vcmi#3721]
**Describe the bug**
When resources are sent to another player, H3 shows notification to a player that receives these resources.

**To Reproduce**
1. Start multiplayer game
2. Open marketplace and send resources to another player
3. Look at game client of player that should receive these resources

**Expected behavior**
Player that received resources should see message box with list of received resources.
Note that in H3 this is not used in hotseat but only in network MP or for resources sent by AI. We can extend this to all MP games if this is more convenient to implement.

**Actual behavior**
Resources are transferred, but there is no notification

**Did it work earlier?**
Not implemented feature

**Additional context**
Migrated from https://bugs.vcmi.eu/view.php?id=2143

**Screenshots**
H3 behavior:
![зображення](https://github.com/vcmi/vcmi/assets/1576820/c3f9877f-dac0-451b-a0ea-b1b3db8fc0a4)

**Comment by Alucard648** [comment]
In OH3 AI in team games on random maps sometimes sends resource to player and then wants other resources in exchange.

**Comment by Laserlicht** [comment]
> In OH3 AI in team games on random maps sometimes sends resource to player and then wants other resources in exchange.

Yes. Found these strings. Do you want to open separate issue? As it is mainly AI issue.

**Comment by Alucard648** [comment]
> > In OH3 AI in team games on random maps sometimes sends resource to player and then wants other resources in exchange.
> 
> Yes. Found these strings. Do you want to open separate issue? As it is mainly AI issue.

Ticket opened.
